Clean/Circomlib/Gates.lean

```diff
@@ -34,7 +34,7 @@ template XOR() {
 def main (input : Expression (F p) × Expression (F p)) := do
   let a := input.1
   let b := input.2
-  let out <== a + b - 2*a*b
+  let out <== a + b - 2 * a * b
```

Comment:
is this really the convention we want? without spaces around `*` it looks closer to what mathematicians would use:

$$
a + b - 2ab
$$